migrations/versions/0441_add_unsubscribe_link.py

```diff
@@ -0,0 +1,29 @@
+"""
+
+Revision ID: 0441_add_unsubscribe_link
+Revises: 0440_new_sms_allowance_n_rate
+Create Date: 2024-03-25 14:38:54.618674
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = "0441_add_unsubscribe_link"
+down_revision = "0440_new_sms_allowance_n_rate"
+
+
+def upgrade():
+    op.add_column("notifications", sa.Column("unsubscribe_link", sa.String(), nullable=True))
+    op.create_check_constraint(
+        "ck_unsubscribe_link_is_null_for_letters_and_sms",
+        "notifications",
+        """
+        notification_type = 'email' OR unsubscribe_link is null
```

Comment:
additionally, can we add a test that asserts that the constraint is correct (by manually allowing both null and non-null emails, while rejecting non-null unsub links?).

this'll also mean we update create_notification fixtures in the db which would be good to do

Reply:
I'm sorry, could you elaborate? I am not sure I understand what you're asking for.

Do you want me to add a unit test that expects error when trying to create an sms/letter with an unsubscribe link?

> this'll also mean we update create_notification fixtures in the db

Do you mean test fixtures here, or something in real database?

I intentionally kept this PR small, as it's a migration PR. I wanted to do more codey things in a next, separate PR. But if you think the constraint needs testing, I could add that?

Reply:
yeah, i think a test where we just make sure the constraint works would be good. it'd be awkward if we accidentally made a mistake in this constraint and haven't tested the code and need to do another migration to fix it.

spitballing, but could do something like
```python
@pytest.mark.parametrise('template_type, unsubscribe_link', [
 ('email', None),
 ('email', 'https://unsub.com'),
 pytest.mark.xfail('sms', 'https://unsub.com', raises=SomeException),
]
def test_constraint_works():
  notification = create_notifications(type=type, unsubscribe_link=unsubscribe_link)
  db.commit()
  assert notification.unsubscribe_link = unsubscribe_link
```

Reply:
Ok, this is done now :)